tests/components/modbus/conftest.py

```diff
@@ -118,8 +122,24 @@ async def base_test(
             async_fire_time_changed(hass, now)
             await hass.async_block_till_done()
 
-        # Check state
         entity_id = f"{entity_domain}.{device_name}"
+
+        # Call an arbitrary service
+        if "call_service" in kvargs:
+            call_service = kvargs["call_service"]
+            await hass.services.async_call(
+                call_service["domain"],
+                call_service["service"],
+                {ATTR_ENTITY_ID: entity_id, **call_service.get("data", {})},
+                blocking=True,
+            )
+            await hass.async_block_till_done()
+
+        # check write_register call
+        if "write_register" in kvargs:
+            args, kvargs = kvargs["write_register"]
+            mock_sync.write_register.assert_called_once_with(*args, **kvargs)
```

Comment:
We should check that the values sent to pymodbus (bytes) correspond the to the parameters in the original call (which could be e.g. STATE_ON).